lib/style_transfer.ex

```diff
@@ -0,0 +1,168 @@
+defmodule Membrane.StyleTransfer do
+  @moduledoc """
+  `Membrane.Filter` transferring style of the video frames using AI model.
+
+  It receives and returns raw video frames in RGB format.
+
+  Available styles are:
+   - `:candy`
+   - `:kaganawa`
+   - `:mosaic`
+   - `:mosaic_mobile`
+   - `:picasso`
+   - `:princess`
+   - `:udnie`
+   - `:vangogh`
+
+  Style can be selected by specyfing `:style` option during spawning a child or by sending `{:set_style, style}` notification from a parent.
+
+  To optimize the element, you can pass some positive integer to `:batch_size` option, but remember that it will icrease the latency.
+  """
+  use Membrane.Filter
+
+  defguard is_pos_integer(term) when is_integer(term) and term > 0
+
+  def_input_pad :input,
+    accepted_format:
+      %Membrane.RawVideo{pixel_format: :RGB, height: height, width: width}
+      when is_pos_integer(height) and is_pos_integer(width)
+
+  def_output_pad :output,
+    accepted_format:
+      %Membrane.RawVideo{pixel_format: :RGB, height: height, width: width}
+      when is_pos_integer(height) and is_pos_integer(width)
+
+  def_options style: [
+                required?: true,
+                spec: style(),
+                description: """
+                Style used by the element.
+
+                Can be changed by sending `{:set_style, style}` notification from a parent.
+                """
+              ],
+              batch_size: [
+                require?: false,
+                default: 1,
+                spec: non_neg_integer(),
+                description: """
+                Number of video frames passed in one batch to the model.
+
+                Can be increased in order to optimize the element at the cost of increased latency.
+
+                Default to 1.
+                """
+              ]
+
+  @styles [:candy, :kaganawa, :mosaic, :mosaic_mobile, :picasso, :princess, :udnie, :vangogh]
+
+  @type style ::
+          unquote(
+            @styles
+            |> Bunch.Typespec.enum_to_alternative()
+          )
+
+  @spec styles() :: [style()]
+  def styles(), do: @styles
```

Comment:
Could we do the following?
```suggestion
  @spec available_styles() :: [style()]
  def available_styles(), do: @styles
```
I start feeling dizzy when I see all those `style`, `@styles`, `style()` etc. :D